src/app/src/main/java/com/example/bookmark/SignupActivity.java

```diff
@@ -0,0 +1,200 @@
+        });
+
+        // allows user to see if their email address is valid as they type
+        phoneNumberEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                checkIfEditTextValidPhoneNumber(phoneNumberEditText, phoneNumberLayout);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+
+        // allows user to see if their phone number is valid as they type
+        emailAddressEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                checkIfEditTextValidEmail(emailAddressEditText, emailAddressLayout);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+        });
+
+        // focus change listeners
+        for (int i = 0; i < formTexts.size(); i++) {
+            EditText text = formTexts.get(i);
+            TextInputLayout layout = formLayouts.get(i);
+            formTexts.get(i).setOnFocusChangeListener(new EmptyTextFocusListener(text, layout));
+        }
+
+        emailAddressEditText.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) {
+                emailAddressLayout.setError(null);
+            } else {
+                if (emailAddressEditText.getText().toString().equals("")) {
+                    validateEditTextEmpty(emailAddressEditText, emailAddressLayout);
+                } else {
+                    checkIfEditTextValidEmail(emailAddressEditText, emailAddressLayout);
+                }
+            }
+        });
+
+        phoneNumberEditText.setOnFocusChangeListener((v, hasFocus) -> {
+            if (hasFocus) {
+                phoneNumberLayout.setError(null);
+            } else {
+                if (phoneNumberEditText.getText().toString().equals("")) {
+                    validateEditTextEmpty(phoneNumberEditText, phoneNumberLayout);
+                } else {
+                    checkIfEditTextValidPhoneNumber(phoneNumberEditText, phoneNumberLayout);
+                }
+            }
+        });
+    }
+
+    /**
+     * Function to abstract away simply adding the user to firestore database
+     *
+     * @param user
+     * @param firebaseProvider
+     */
+    private void addUser(User user, FirebaseProvider firebaseProvider) {
+        firebaseProvider.storeUser(user,
+            aVoid -> {
+            },
+            e -> Toast.makeText(SignupActivity.this, "Connection Error. Please Try again.", Toast.LENGTH_LONG).show());
+    }
+}
```

Comment:
Because of the scenario outlined in my previous comment, this method should really be deleted.